macos/FluentUI/Button/Button.swift

```diff
@@ -131,6 +131,42 @@ open class Button: NSButton {
 		}
 	}
 
+	/// When set to a non `nil` value ,this image will always be placed on the trailing edge of the button.
+	/// It is designed to remain independent of  the `imagePosition` value of the NSButton which applies to the built-in `image` property.
+	@objc public var trailingImage: NSImage? {
+		didSet {
+			guard oldValue != trailingImage else {
+				return
+			}
+
+			guard let cell = cell as? ButtonCell else {
+				preconditionFailure("The FluentUI `Button` should only be used with the `ButtonCell` cell class.")
+			}
+
+			// Handle replacement/removal of a previously set image
+			trailingImageView?.image = nil
```

Comment:
Can do this in a corresponding `else` block for the `if` condition on line 149

Reply:
And then we should also remove it from our view hierarchy and nil it out in that case